File: trading-platform/src/main/java/com/tradingplatform/tradingplatform/rate/Rate.java
package com.tradingplatform.tradingplatform.rate;



import com.tradingplatform.tradingplatform.shared.CryptoCurrency;

import java.math.BigDecimal;
import java.time.LocalDateTime;


public record Rate(CryptoCurrency currency, BigDecimal value, LocalDateTime timestamp) {
}

File: exchange/src/main/kotlin/com/trader/exchange/Rate.kt
package com.trader.exchange

import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal
import java.time.LocalDateTime

data class RatesResponse(
    @JsonProperty("asset_id_base")
    val assetIdBase: String,
    val rates: List<RateDetails>
)

data class RateDetails(
    @JsonProperty("asset_id_quote")
    val assetIdQuote: String,
    val time: LocalDateTime,
    val rate: BigDecimal
)
